### DISCUSSION TITLE
How to rename a vtable function

### DISCUSSION THREAD

**Discussion by wjllz** [Vector35/binaryninja-api#3035] [Q&A] [answered]
Hey, I have some code like this:

``` c++
// original source code
OSObject     * o;
o->release();
```
So I create a structure in binary ninja:
![image](https://user-images.githubusercontent.com/28553622/159428980-433e57b2-24cb-4a94-bb0f-237daa059cd6.png)
It will decompile it to :
![image](https://user-images.githubusercontent.com/28553622/159429046-37e23cae-9d14-476c-a2c9-5c3849387445.png)
Is there any way to make it be "o-relase()"? 

thx and sry for my bad English...

**Comment by CouleeApps** [comment, marked as answer]
Chances are, your vtable is defined as a pointer to a structure containing function pointers. You will need two structures: one for the vtable and one for the object. In the object structure, the first member should be a pointer to a vtable structure. In the vtable structure, you should define function pointers at the offsets where they are used.

In your example, it would probably look like this:
```
struct OSObject_vtable
{
    char unknown[0x28]; // ~5 more functions here that are not known
    int64_t* (*release)(struct OSObject* this);
};
struct OSObject
{
    OSObject_vtable* vtable;
    // Rest of fields of OSObject
};
```

**Reply by wjllz** [reply]
shame of myself! thank u so much!
